source/setup.c

```diff
@@ -216,30 +233,67 @@ u32 setup_main(void* arg){
     update_error_state(ret, 1);
 
     // write out log entries from before the SD was mounted
-    write_log(fsaHandle, logHandle, "MakeQuota",  "/vol/storage_mlc01/sys", sys_quota_ret);
+    log_printf(fsaHandle, logHandle, "MakeQuota",  "/vol/storage_mlc01/sys", sys_quota_ret);
     for(i = 0; folders_to_create[i]; i++){
-        write_log(fsaHandle, logHandle, "MakeDir", folders_to_create[i], folder_ret[i]);
+        log_printf(fsaHandle, logHandle, "MakeDir", folders_to_create[i], folder_ret[i]);
     }
-    write_log(fsaHandle, logHandle, "Flush", "MLC", flush_ret);
+    log_printf(fsaHandle, logHandle, "Flush", "MLC", flush_ret);
 
     install_all_titles(fsaHandle, "/vol/sdcard/wafel_install", logHandle);
     flush_ret = flush_mlc(fsaHandle);
     update_error_state(flush_ret, 2);
-    write_log(fsaHandle, logHandle, "Flush", "MLC", flush_ret);
+    log_printf(fsaHandle, logHandle, "Flush", "MLC", flush_ret);
+
+    // Massive props to Gary
+    int mcp_handle = iosOpen("/dev/mcp", 0);
+
+    MCPSysProdSettings sysProdSettings;
+    ret = MCP_GetSysProdSettings(mcp_handle, &sysProdSettings);
+    debug_printf("MCP_GetSysProdSettings: %X\n", ret);
+    if (ret != 0) {
+      // handle failure to not corrupt
+      return -1;
+    }
+
+    int region = -1;
+    int dirHandle = -1;
+    if(ret = FSA_OpenDir(fsaHandle, "/vol/storage_mlc01/sys/title/00050010/10040000", &dirHandle) == 0){
+        region = MCP_REGION_JAPAN;
+    } else if(ret = FSA_OpenDir(fsaHandle, "/vol/storage_mlc01/sys/title/00050010/10040100", &dirHandle) == 0){
+        region = MCP_REGION_USA;
+    } else if(ret = FSA_OpenDir(fsaHandle, "/vol/storage_mlc01/sys/title/00050010/10040200", &dirHandle) == 0){
+        region = MCP_REGION_EUROPE;
+    } else {
+        // handle failure
+        return -1;
```

Comment:
don't crash the whole setup, just give a warning by changing the LED, write it to log and don't do the sysprod stuff